cpp/ql/src/semmle/code/cpp/Type.qll

```diff
@@ -795,10 +946,38 @@ class Decltype extends Type, @decltype {
   }
 }
 
+/**
+ * A C/C++ pointer or reference type.
+ */
+class PointerIshType extends DerivedType {
```

Comment:
I expect this class to go away from the public-visible hierarchy when the four overridable helper functions are removed.

Reply:
It seems like it would be generally useful, and consistent with `FunctionPointerIshType`. Admittedly, I don't have a real use case for it yet other than in the `Print` module, so I've removed it from the public API. It's easy enough to add back in later if we do find a real use for it.